### DISCUSSION TITLE
How can I convert jpeg to jpeg-xl (lossless)

### DISCUSSION THREAD

**Discussion by JoeWierer** [dlemstra/Magick.NET#872] [Help] [answered]
Hi.
Can somebody give me an example (or some c# sourcecode) how to convert jpeg to jpeg-xl (lossless) with Magick.NET?
(jpeg-xl should be ~20% smaller than the jpeg)

**Comment by dlemstra** [comment]
You will need to set the `Quality` of the image to 100 for losless compression. And I think it depends on the type of image if the file will be smaller.

**Comment by ghost** [comment, marked as answer]
ImageMagick currently does **not** support JPEG XL's lossless JPEG transcode feature.

If you want to use JPEG XL's lossless JPEG transcode feature, and be able to get back the original JPG file, you have to use the [reference software](https://gitlab.com/wg1/jpeg-xl).

Even if you set ImageMagick's quality to 100, IM will decode the JPG to pixels and encode the pixels losslessly to JXL. However, this is actually slightly lossy (see this [Reddit comment](https://old.reddit.com/r/jpegxl/comments/n36kk0/would_the_result_of_jpeg_png_jpeg_xl_be_larger/gwnupqp/) by JPEG XL developer for explanation).

To sum up:
Since IM does not support this feature yet, for the time being you can use reference software, encode with `cjxl image.jpg image.jxl`, enjoy ~20% smaller size, and you can get back the original JPG file (bit-by-bit the same, thus same file checksum) with `djxl image.jxl image.jpg`.

**Reply by JoeWierer** [reply]
This is the information I was looking for ... Thanks a lot.

**Reply by Zatigemm** [reply]
To tell `cjxl` to encode in lossless mode, you need the `-quality 100` option. And you can encode in lossless in magick with `-quality 100` as well… You may see a very small difference when you decode to get the original image from JXL, but this is just the metadata of the file that has changed, some additional info at the head of it…

**Reply by drhead** [reply]
> To tell `cjxl` to encode in lossless mode, you need the `-quality 100` option. And you can encode in lossless in magick with `-quality 100` as well… You may see a very small difference when you decode to get the original image from JXL, but this is just the metadata of the file that has changed, some additional info at the head of it…

You don't have to specify quality when using `cjxl` to convert JPEGs, it will default to lossless transcoding, mainly because this is the only remotely practical option for transcoding most JPEGs.  So the above example works as-is.
